5.2.cup_of_join.py

```diff
@@ -0,0 +1,35 @@
+def join(*param, sep='-'):
+    # use a list comprehension to flatten the lists
+    joined_list = []
+
+    # add the separator between the items
+    for lst in param:
```

Comment:
Read this - https://stackoverflow.com/questions/716477/join-list-of-lists-in-python
and this - https://www.programiz.com/python-programming/methods/built-in/zip
And think if you can make your code shorter